### ISSUE TITLE
[Dropdown] Default value is not working when it is "0"

### ISSUE THREAD

**Issue by ultmaster** [Semantic-Org/Semantic-UI#5638]
JSFiddle: https://jsfiddle.net/b0xa4pzz/2/

There are two problems.

First, `0` does not work as a default value. This is probably caused by "fallback to false". I am not sure whether this has a good solution. I managed to solve this problem by add `v` to `0`, becoming `v0`. But I was hoping someone can propose something less brute-force.

This problem is shown in Example 1 and 2.

Second, it seems that all integers does not work as a default value. **I failed to demonstrate that on Fiddle, because I was struggling to use AJAX. Hope someone can help me with that.** The 3rd example will not work and the 4th example works.

I believe I have tracked down the cause of this problem: https://github.com/ultmaster/Semantic-UI/blob/master/src/definitions/modules/dropdown.js#L1770 . By comparing `values` to `string`, we have missed the case in which `values` is a number, etc. I changed that into `values != 'object'`, and everything was working perfectly.

I believe that there is something we can do to solve both problems. **That is convert the value into string when we first get the default values.** 

So, is there somebody who can help me with my Fiddle?

**Comment by Banandrew** [comment]
Hi @ultmaster, thank you, we’re already tracking the first problem here: #5354. As for the second problem, the dropdown component blows up, because your markup is incorrect, [and even if it was](https://semantic-ui.com/modules/dropdown.html#/usage), it wouldn’t be able to select a value, because a dropdown’s menu would be empty, and it’s not empty only when an endpoint is hit; we’re tracking it as enhancement here: #5625.

I don’t think that `.ui.search.dropdown` is supported, so let’s use the `.selection` variation instead:
```HTML
<div class="ui search selection dropdown">
  <input type="hidden">
  <i class="dropdown icon"></i>
  <input class="search" autocomplete="off">
  <div class="default text">Select me</div>
  <div class="menu"></div>
</div>
```
And to mock your `data`:
```JS
$('.ui.dropdown.search')
  .dropdown({
    apiSettings: {
      response: data
    }
  })
```
`module.get.remoteValues()` doesn’t cover this case, because it simply gets values from `sessionStorage` when a page is refreshed to set them again, and has nothing to do with the issue—it all comes down to if `module.set.selected()` should work with remote content (#5354).

If you have more questions, I’ll be happy to answer them in this thread.

**Comment by lubber-de** [comment]
Default zero issue is fixed in [fomantic/Fomantic-UI#370](https://github.com/fomantic/Fomantic-UI/pull/370)
I updated your fiddle accordingly here: https://jsfiddle.net/b0xa4pzz/3/
